cdc/sink/codec/avro.go

```diff
@@ -522,3 +525,69 @@ func (r *avroEncodeResult) toEnvelope() ([]byte, error) {
 	}
 	return buf.Bytes(), nil
 }
+
+type avroEventBatchEncoderBuilder struct {
+	credential *security.Credential
+	opts       map[string]string
+
+	keySchemaManager   *AvroSchemaManager
+	valueSchemaManager *AvroSchemaManager
+
+	initialized *atomic.Bool
+}
+
+func newAvroEventBatchEncoderBuilder(credential *security.Credential, opts map[string]string) EncoderBuilder {
+	return &avroEventBatchEncoderBuilder{
+		credential:  credential,
+		opts:        opts,
+		initialized: atomic.NewBool(false),
+	}
+}
+
+const (
+	keySchemaSuffix   = "-key"
+	valueSchemaSuffix = "-value"
+)
+
+// newSchemaManagers initialize the `keySchemaManager` and `valueSchemaManager` in an atomic way.
+func (b *avroEventBatchEncoderBuilder) newSchemaManagers(ctx context.Context) error {
+	registryURI, ok := b.opts["registry"]
+	if !ok {
+		return cerror.ErrPrepareAvroFailed.GenWithStack(`Avro protocol requires parameter "registry"`)
+	}
+
+	keySchemaManager, err := NewAvroSchemaManager(ctx, b.credential, registryURI, keySchemaSuffix)
+	if err != nil {
+		return err
+	}
+	b.keySchemaManager = keySchemaManager
+
+	valueSchemaManager, err := NewAvroSchemaManager(ctx, b.credential, registryURI, valueSchemaSuffix)
+	if err != nil {
+		return err
+	}
+	b.valueSchemaManager = valueSchemaManager
+
+	return nil
+}
+
+// Build an AvroEventBatchEncoder.
+func (b *avroEventBatchEncoderBuilder) Build(ctx context.Context) (EventBatchEncoder, error) {
+	if !b.initialized.Load() {
+		if err := b.newSchemaManagers(ctx); err != nil {
+			return nil, cerror.WrapError(cerror.ErrPrepareAvroFailed, err)
+		}
+		b.initialized.Store(true)
+	}
+
+	encoder := NewAvroEventBatchEncoder().(*AvroEventBatchEncoder)
```

Comment:
`NewAvroEventBatchEncoder` doesn't need to be exposed now, and it can return `*AvroEventBatchEncoder` directly